### DISCUSSION TITLE
Flashing sonoff zigbee bridge did not complete = bricked?

### DISCUSSION THREAD

**Discussion by jenshenk** [arendst/Tasmota#12398] [Q&A] [answered]
Hi,
I would be very thankful to get some input on this: After having flashed a rf bridge with no problems and integrated some rf stuff  I wanted to also dive into the zigbee ecosystem. After opening the zigbee bridge I noticed the smaller holes and since I did not want to wait on ordering something I McGuyverd it by cutting off the ends of dupont female connectors and just sticked the cables into the holes... Following a guide I connected GND-GPIO0 and the rest as usual
To my own surprise it actually worked and started flashing (using tasmotizer), yet half way through it disconnected :( the only thing that worked since then was getting through the erase phase but now I just can't connect anymore...

1. Is the bridge bricked now or is there still hope?
2. Is it normal that with this device the green led (labeled zigbee) is lid in flashing mode?
3. And maybe a real NOOB question: How do I actually know that the connection from the adapter to the board works? I seem to have bad luck with the dupont cables I order and in the past I had one instance where after a lot of tries I pushed a little bit on the top of the connector (where the cable enters the plug) and then it finally worked. 

Thanks so much for any guidance!

**Comment by sfromis** [comment, marked as answer]
My guess is that your wiring with just sticking pins into holes works badly. When I use that inferior method, I make sure to apply a firm sideways finger pressure to the susceptible pins while I want them to be working. That improves odds, but I may still have to retry.

**Reply by jenshenk** [reply]
Thanks for the quick reply! Makes sense... maybe it works just a little bit better with bigger holes... that way I was able to flash a couple of S20s and even put openwrt on a router
So I guess I can still rescue it but I need to seriously up my soldering skills... damn, looked so easy in the youtube vids w/o soldering :)

**Reply by NsinghP** [reply]
I don’t know if this is relevant but I ran into similar problem when flashing couple of ESP8266 12F chips. It turned out that the USB-to-TTL adapter was faulty as I had fried/shorted it out somehow when connecting the wires. I was able to get around the problem by using spare NodeMCU to flash the chips. 

EDIT: in my case, after unsuccessful flash I started playing around with different options and the chip would erase properly and flashing phase would stop at 8% or 20% or even at 90% depending on baud rate  and options (DIO etc. I was using) in the end it was not the settings or wire connection, it was faulty adapter.

**Reply by jenshenk** [reply]
Thanks for taking the time to share this, indeed something to keep in my mind that even these simple things can die. Unfortunately, I have already tried two different ones and just to rule out hardware issues flashed another device (with bigger holes to connect though) with one of them which worked flawlessly. So I guess the only solution remains soldering - even more since I saw with a magnifying glass that all that tinkering around led to scrape the connectors a bit which isn't helping. Nevertheless, very nice to know that even when flashing goes wrong a couple of times the devices are not bricked so thanks for sharing your experience!

**Reply by sfromis** [reply]
What I like to do for flashing the Sonoff ZbBridge is to solder tiny headers to the small holes, that makes the risk of ripping traces lower. I've used the metal part from inside a dupont female heder (works for plugging a female dupont into), but just a bit of solid wire from a network cable could also do nicely.

**Reply by sfromis** [reply]
Note: In the Discord server, we've seen a few cases of people getting a ZbBridge where the conclusion was that the flash chip was bad, it helped to replace.

**Reply by jenshenk** [reply]
Seems I now have a clear plan how to proceed... thanks so much guys!
